### PR TITLE
VZ-5230: Initial change for alternate test framework

### PR DESCRIPTION
**Pull request by bhatpmk**
# Description

This PR contains initial changes for the alternate test framework, inspired by Kubernetes framework for end-to-end testing.

Contributes to VZ-5230

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

File: tests/e2e/framework/metrics.go [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+	}
+
+	branchName := os.Getenv("BRANCH_NAME")
+	if branchName != "" {
+		log = log.With(BranchName, branchName)
+	}
+
+	buildURL := os.Getenv("BUILD_URL")
+	if buildURL != "" {
+		buildURL = strings.Replace(buildURL, "%252F", "/", 1)
```
Comment (markxnelson) [inline tests/e2e/framework/metrics.go:114]: use urlencode instead

File: tests/e2e/framework/search_writer.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+	return SearchWriter{
+		hc:    retryablehttp.NewClient(),
+		url:   uri,
+		index: index,
+		auth:  auth,
+	}, nil
+}
+
+//Close implement as needed
```
Comment (markxnelson) [inline tests/e2e/framework/search_writer.go:47]: what does this comment mean?

File: tests/e2e/framework/search_writer.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	req, err := retryablehttp.NewRequest("POST", uri, reader)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	if basicAuth != "" { // Basic Auth is used if provided
+		req.Header.Set("Authorization", fmt.Sprintf("basic %s", basicAuth))
+	}
+	resp, err := hc.Do(req)
```
Comment (markxnelson) [inline tests/e2e/framework/search_writer.go:69]: will this actually retry?

**Review by markxnelson — COMMENTED** [review]
see comments

File: tests/e2e/framework/metrics.go (hunk 1 above)
Comment (bryantidd) [reply tests/e2e/framework/metrics.go:114]: same

File: tests/e2e/framework/search_writer.go (hunk 2 above)
Comment (bryantidd) [reply tests/e2e/framework/search_writer.go:47]: I believe it's a reminder to close client connections as needed.

File: tests/e2e/framework/search_writer.go (hunk 3 above)
Comment (bryantidd) [reply tests/e2e/framework/search_writer.go:69]: no, just being consistent with the client

**Review by markxnelson — APPROVED** [review]
approved on the understanding this is not the final pr, but a step on the way there

**Pull request merged**